packages/cln_plugin/lib/src/json_rpc/request.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'request.g.dart';
+
+@JsonSerializable()
+class Request {
+  String jsonrpc;
+  // The spec allows to set the id as a String, but core lightning uses int,
+  // so we strictly use int to make the parsing less painful.
+  int id;
+  String method;
+
+  /// TODO add docs comments
+  Map<String, dynamic> params;
+
+  Request(
```

Comment:
Fields `result` and `error` are sort of conditional, how should we define in that case?